File: A/array_with_odd_sum.c
#include <stdio.h>

void solve() {
    int n, a;
    scanf("%d", &n);
    int odd = 0;
    for (int i = 0; i < n; ++i) {
        scanf("%d", &a);
        odd += a % 2;
    }
    if (odd == 0 || (odd == n && !(n % 2))) {
        printf("NO\n");
    } else {
        printf("YES\n");
    }
}

int main(void) {
    int t;
    scanf("%d", &t);
    while (t--) {
        solve();
    }
    return 0;
}

File: A/array_with_odd_sum.py
import sys
input = sys.stdin.readline

def solve():
    n = int(input())
    a = list(map(int, input().split()))
    odd = sum(x % 2 for x in a)
    if odd == 0 or (odd == n and n % 2 == 0):
        print("NO")
    else:
        print("YES")

for _ in range(int(input())):
    solve()
